colin-api/src/colin_api/models/reset.py

```diff
@@ -239,3 +239,49 @@ def reset_filings(cls, start_date: str = None, end_date: str = None, identifiers
             if con:
                 con.rollback()
             raise err
+
+    @classmethod
+    def reset_filings_by_event(cls, event_ids: list = []):
+        """Reset changes made for given event ids."""
+        # initialize reset object
+        reset_obj = Reset()
+
+        # place into lists that can be reset together
+        annual_report_events = []
+        events_info = []
+
+        for filing_info in reset_obj.get_filings_for_reset():
+            if filing_info['event_id'] in event_ids:
+                events_info.append(filing_info)
+                if filing_info['filing_typ_cd'] in Filing.FILING_TYPES['annualReport']['type_code_list']:
+                    annual_report_events.append(filing_info['event_id'])
+
+        try:
+            if event_ids:
+                # setup db connection
+                con = DB.connection
+                con.begin()
+                cursor = con.cursor()
+
+                # reset data in oracle for events
+                # The commented out events do not seem to happen for AR so they are commented out.
+                new_corps = cls._get_incorporations_by_event(cursor, event_ids)
+                #Party.reset_dirs_by_events(cursor=cursor, event_ids=event_ids)
+                #Office.reset_offices_by_events(cursor=cursor, event_ids=event_ids)
+                #Business.reset_corp_states(cursor=cursor, event_ids=annual_report_events)
+                Business.reset_corporations(cursor=cursor, event_info=events_info, event_ids=event_ids)
+                #ShareObject.delete_shares(cursor, event_ids)
+                cls._delete_filing_user(cursor=cursor, event_ids=event_ids)
+                #cls._delete_ledger_text(cursor=cursor, event_ids=event_ids)
+                #cls._delete_corp_name(cursor=cursor, event_ids=list(new_corps.values()))
+                cls._delete_corp_state(cursor=cursor, corp_nums=list(new_corps.keys()))
```

Comment:
I'm not entirely sure about this but I'm assuming there is a need to delete all corp state entries that exists for corps that had filed an AR from BAR?

Just confirming that this is needed.

Reply:
That one does indeed generate a state change, as they get marked as having filed their AR so without that it'll continue to think it was filed.

Reply:
ok, resolving.